src/plugins/vis_type_timeseries/public/application/components/data_source_picker.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+
+import { SavedObjectsClientContract, ToastsStart } from 'src/core/public';
+import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
+import { DATA_SOURCE_ID_KEY } from '../../../common/constants';
+
+/**
+ * Provide only the necessary plugin setup/start and dataSourceId and this component will render a DataSourceSelector with a default selection
+ *
+ * @property {SavedObjectsClientContract} savedObjectsClient
+ * @property {DataSourceManagementPluginSetup} dataSourceManagement
+ * @property {ToastsStart} toasts
+ * @property {string} [defaultDataSourceId] - the datasource id as the default option when the component first renders
+ * @property {(e: Array<{}>) => void} handleChange - the function that will update the model when a datasource is selected
+ */
+export interface DataSourcePickerProps {
+  savedObjectsClient: SavedObjectsClientContract;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+  toasts: ToastsStart;
+  defaultDataSourceId?: string;
+  handleChange: (e: Array<{}>) => void;
+}
+
+/**
+ * Provides a wrapper around the DataSourceSelector component exposed by core
+ *
+ * @param {DataSourcePickerProps} props
+ * @returns the DataSourceSelector
+ */
+export const DataSourcePicker = (props: DataSourcePickerProps) => {
+  const { savedObjectsClient, defaultDataSourceId, handleChange } = props;
+  const [defaultOption, setDefaultOption] = useState<Array<{ id: string; label: string }>>();
+  const DataSourceSelector = props.dataSourceManagement.ui.DataSourceSelector;
+
+  const onDataSourceSelectChange = (dataSourceOption: Array<{ id: string; label: string }>) => {
+    setDefaultOption(dataSourceOption);
+    handleChange(dataSourceOption);
+  };
+
+  useEffect(() => {
+    if (!defaultDataSourceId || defaultDataSourceId === '') {
+      // @ts-expect-error
+      setDefaultOption(null);
```

Comment:
Looks like we use:
1. real dataSourceId + label: as default dataSource option
2. undefined: as page is still loading
3. null: as no default dataSource.
And ```setDefaultOption(null);``` will cause ts-lint error.

I'm thinking instead of setting defaultOption, what if we only set defaultDataSourceId, and DataSourceSelector receive defaultDataSourceId instead of defaultDataSourceOption. And in DataSourceSelector, we grab the Label?

In that case we can avoid some potential localization issue in the future (Different language switch).

What do you think?

Reply:
Yes that is a good idea, and @yujin-emma has raised a PR to partially address this (#6315). That PR does not implement the `DataSourceSelector` component so for the time being, this PR implements a workaround.